app/frontend/src/submissions/components/SubmissionForm/Owner.vue

```diff
@@ -135,14 +145,108 @@ export default {
+     * On failure, it logs the error and clears the current suggestions.
+     * Finally, sets the loading state to false.
+     */
+    async fetchAddressSuggestions() {
+      if (!this.ownerAddressInput) {
+        this.addressSuggestions = [];
+        return;
+      }
+      this.isLoadingSuggestions = true;
+      const params = {
+        minScore: 50, //accuracy score of results compared to input
+		    maxResults: 5,
+        echo: 'false',
+        brief: true,
+        autoComplete: true,
+        addressString: this.ownerAddressInput
+      };
+
+      const querystring = require('querystring');
+      const searchParams = querystring.stringify(params);
+      try {
+        const response = await fetch(`${GEOCODER_ADDRESS_API}${searchParams}`);
+        const data = await response.json();
+        if (data && data.features) {
+          
+          this.addressSuggestions = data.features.map(item => item.properties.fullAddress);
+        } else {
+          this.addressSuggestions = [];
+        }
+      } catch (error) {
+        console.error(error);
+        this.addressSuggestions = [];
+      } finally {
+        this.isLoadingSuggestions = false;
+      }
+    },
+
+    /**
+     * Processes the selected address suggestion.
+     * Splits the suggestion into components and updates the owner's province, city, and address inputs accordingly.
+     * Clears the address suggestions afterward.
+     * @param {string} suggestion - The selected address suggestion. ("1234 Street Rd, Name of City, BC")
+     */
+    selectAddressSuggestion(suggestion) {
+      const ownerAddressArray = suggestion.split(',');
+      if(ownerAddressArray){
+        if(ownerAddressArray[ownerAddressArray.length -1].toUpperCase().trim() === 'BC' || ownerAddressArray[-1].toUpperCase().trim() === 'BRITISH COLUMBIA')
+        {
+          this.ownerProvinceInput = this.codes.province_codes[0].province_state_code;
+          this.ownerAddressInput = '';
+        }
+      else this.ownerProvinceInput = "";
+      this.ownerCityInput = ownerAddressArray[ownerAddressArray.length -2];
+      if(ownerAddressArray[ownerAddressArray.length -3]) this.ownerAddressInput = ownerAddressArray[ownerAddressArray.length -3];
+      }
+      
+      this.clearAddressSuggestions();
+    },
+
+    /**
+     * Clears the current list of address suggestions.
+     */
+    clearAddressSuggestions () {
+      this.addressSuggestions = [];
+    },
+
+    /**
+     * Shows the address suggestions list in the UI.
+     */
+    showList() {
+      if(document.getElementById('address-suggestions-list')){
+        document.getElementById('address-suggestions-list').style.display = 'block';
+      }     
+    },
+
+    /**
+     * Hides the address suggestions list in the UI.
+     */
+    hideList() {
+      if(document.getElementById('address-suggestions-list')){
+        document.getElementById('address-suggestions-list').style.display = 'none';
+      }
+    }     
```

Comment:
```suggestion
    showList(show) {
      if(document.getElementById('address-suggestions-list')){
        document.getElementById('address-suggestions-list').style.display =  show? 'block' : 'none';
      }     
    },
```
Couldn't this be reduced to one function taking a boolean, and reduce code duplication / overall complexity?

Reply:
implemented